aea/aea_builder.py

```diff
@@ -919,8 +907,30 @@ def _load_connection(address: Address, configuration: ConnectionConfig) -> Conne
             address=address, configuration=configuration
         )
     except Exception as e:
-        raise Exception(
-            "An error occurred while loading connection {}: {}".format(
-                configuration.public_id, str(e)
-            )
+        _handle_error_while_loading_component_generic_error(configuration, e)
+    assert False  # this is to make MyPy stop complaining of "Missin return statement".
+
+
+def _handle_error_while_loading_component_module_not_found(
+    configuration: ComponentConfiguration, e: ModuleNotFoundError
```

Comment:
above you're filtering for all exceptions. but here you limit it to ModuleNotFoundError..

Reply:
this handler will be called in the `except ModuleNotFoundError` clause which I'll add in a next commit

Reply:
of course, since `ModuleNotFoundError` is-a `Exception`, it must be before `except Exception` so it will be matched before `Exception`